File: 3267-find-longest-special-substring-that-occurs-thrice-i/3267-find-longest-special-substring-that-occurs-thrice-i.py
class Solution:
    def findTwice(self, s, special, ind):
        cnt = 2
        n = len(s)
        length = len(special)
        for j in range(ind, n - length + 1):
            if s[j:j + length] == special:
                cnt -= 1
                if cnt == 0:
                    return True
        return False

    def maximumLength(self, s):
        n = len(s)
        ans = -1

        for i in range(n):
            special = ""
            for j in range(i, n):
                if special and special[-1] != s[j]:
                    break
                special += s[j]
                if self.findTwice(s, special, i + 1):
                    ans = max(ans, len(special))

        return ans

File: 3267-find-longest-special-substring-that-occurs-thrice-i/3267-find-longest-special-substring-that-occurs-thrice-i.cpp
class Solution {
public:
    bool findTwice(string &s, string &special, int ind)
    {
        int cnt = 2;
        int n = s.size();
        int len = special.size();
        for(int j=ind; j<=n-len; j++)
        {
            string sub = s.substr(j, len);
            if(special == sub)
            {
                cnt--;
                if(cnt == 0)
                    return true;
            }
        }
        return false;
    }

    int maximumLength(string s) {
        int n = s.size();
        int ans = -1;

        for(int i=0; i<n; i++)
        {
            string special = "";
            for(int j=i; j<n; j++)
            {
                if(special.empty() == false && special.back() != s[j])
                    break;
                special.push_back(s[j]);

                if(findTwice(s, special, i+1))
                {
                    ans = max(ans, (int)special.size());
                }
            }
        }

        return ans;
    }
};